app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt

```diff
@@ -87,9 +88,13 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
     override fun onStart() {
         super.onStart()
         // https://github.com/mozilla-mobile/fenix/issues/14279
-        // To prevent GeckoView from resizing we're going to change the softInputMode to not adjust
-        // the size of the window.
-        requireActivity().window.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_NOTHING)
+        // This will need to be handled for the update to R. We need to resize here in order to
+        // see the whole homescreen behind the search dialog.
+        @Suppress("DEPRECATION")
+        requireActivity().window.setSoftInputMode(
+            WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE
```

Comment:
See comment above. I don't think we can do an unconditional resize here as we'd regress: https://github.com/mozilla-mobile/fenix/issues/14279

Reply:
The fix for #14279 was reverted, so it hasn't actually been fixed yet. I want to chat with Codrut and figure out how this can be fixed. I think we might be able to "fake" this with a large margin underneath collections on home, but it might look weird.

Reply:
@eliserichards OK let's profile together. I think "faking" it in the UI is still better than any large page load regression, so let's see what we find.